### PR TITLE
fix(models): break autograd chain on energies buffer in Ewald + PME

### PR DESCRIPTION
**Pull request by Ryan-Reese**
# ALCHEMI Toolkit Pull Request

## Description

`EwaldModelWrapper` and `PMEModelWrapper` keeps a persistent `self._energies_buf` tensor and reuses it each forward via `zero_()` + `scatter_add_(0, batch_idx, per_atom_energies)`. Because `per_atom_energies` carries a `grad_fn` from the Warp backward tape, the in-place ops chain each step's autograd graph onto the buffer through PyTorch's version-counter mechanism. The buffer is owned by `self`, so the chain cannot be GC'd — per-step cost grows linearly and GPU memory climbs unboundedly in long MD runs. This PR calls `detach_()` after the clone to break the autograd chain.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Performance improvement
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Related Issues

N/A — bug surfaced during downstream MD integration work, not tracked in an existing issue.

## Changes Made

- `nvalchemi/models/ewald.py`, `nvalchemi/models/pme.py`: for `self._energies_buf`, zero buffer per `forward()`, scatter into it, call `detach_()` after the clone, so callers continue to receive a tensor with independent storage (matches the prior output contract; some downstream consumers mutate `batch.energy` in place).
- `test/models/test_ewald.py`, `test/models/test_pme.py`: add two regression tests per wrapper — one checks `_energies_buf.grad_fn is None`, one asserts consecutive forwards return energy tensors whose storage is independent of prior outputs.
- `CHANGELOG.md`: add entry under Unreleased.

## Testing

- [x] Unit tests pass locally
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

### Empirical verification

per_step_runtime_and_gpu_mem

1000 NVT Langevin steps, naphthalene (3,6,3) = 1944 atoms, PBC, AIMNet2 + electrostatics (`accuracy=1e-6`, `hybrid_forces=False`, `compile_model=True`), dt = 1.5 fs, T = 200 K, friction = 0.01, 50-step timing blocks.

Tested on: NVIDIA H100 NVL (95 GB, driver 570.195.03), ARM (Linux 6.8.0), PyTorch 2.11.0+cu130 (CUDA 13.0), Python 3.12.13.

| | step 50 ms/step | step 950 ms/step | last / first | GPU mem start | GPU mem end |
|---|---|---|---|---|---|
| Ewald (unfixed) | 146.5 | 456.9 | 3.1× | 175.8 MB | 196.5 MB |
| Ewald (fixed)   | 94.4  | 79.7  | 0.84 (warmup) | 174.7 MB | 174.7 MB |
| PME (unfixed)   | 111.8 | 568.6 | 5.1× | 183.9 MB | 204.6 MB |
| PME (fixed)     | 91.6  | 76.0  | 0.83 (warmup) | 182.8 MB | 182.8 MB |

Figure (per-step runtime + GPU memory vs step, all four configs) attached below.

**Numerical equivalence.** GPU `scatter_add_` is non-deterministic (atomic-add ordering), so neither variant is bit-exact against itself. Measured relative L2 deltas of per-atom positions and forces at step 20 (seed = 42): unfixed-vs-fixed deltas are within same-code run-to-run envelope for both models. No systematic shift.

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [x] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [x] I have added docstrings to new functions/classes
- [ ] I have updated the documentation (if applicable)

## Additional Notes

`nvalchemi/models/lj.py` has a superficially similar pattern but with `self._atomic_energies_buf` (a kernel output buffer) on the scatter.

### CODE REVIEW

File: test/models/test_ewald.py [hunk 1]
```diff
@@ -648,6 +648,17 @@ def test_neutral_system_forces_symmetry(self):
...
+    def test_energies_buffer_detached_after_forward(self):
+        """`_energies_buf` has no `grad_fn` after a grad-carrying forward (#82)."""
+        w = _make_ewald()
+        batch = _make_charged_batch()
+        batch.charges = batch.charges.detach().requires_grad_(True)
+        self._build_nl(batch, w)
+        out = w(batch)
+        assert out["energy"].grad_fn is not None
+        assert w._energies_buf.grad_fn is None
+        assert not w._energies_buf.requires_grad
```
[the remark this answers is not in the document]
Comment (Ryan-Reese) [reply test/models/test_ewald.py:660]: @greptileai check again

[the remark this answers is not in the document]
Comment (Ryan-Reese) [reply test/models/test_ewald.py:660]: @greptileai check again

[the remark this answers is not in the document]
Comment (Ryan-Reese) [reply test/models/test_ewald.py:660]: @greptileai check again

**Comment by zubatyuk** [comment]
LGTM

**Review by dallasfoster — APPROVED** [review]
